[All] Add forward declaration and opaque API for ExecParams, MechanicalParams, VisualParams and ConstraintParams

damienmarchal opened the pull request:
ExecParams, ConstraintParams, MechanicalParams and VisualParams are used in a lot of different place of our code base which thus requires to #include their corresponding header causing code coupling. 

This PR contains:
- the implementation of a set of "opaque/c-api" function to be used in header when a ExecParams  (or other Params) is needed. 
  the advantage of using these "opaque" function is that there is not includes other than the "fwd.h".
- the removale of all the #include<XXXXXXParams.h> everywhere is was possible, 
- the adding of the missing #include<XXXXXXParams.h> (or other include) in translation units (.cpp) or (.inl) when it was needed. 

Following our guidlines, the declaration for the "opaque functions" are in ```sofa/core/fwd.h``` and contains:
```cpp
namespace sofa::core
{
namespace execparams
    {
        SOFA_CORE_API ExecParams* defaultInstance();
    }

    namespace constraintparams
    {
        SOFA_CORE_API const ConstraintParams* defaultInstance();
        SOFA_CORE_API ExecParams* asExecParams(sofa::core::ConstraintParams*);
        SOFA_CORE_API const ExecParams* asExecParams(const sofa::core::ConstraintParams*);
    }

    namespace mechanicalparams
    {
        SOFA_CORE_API const MechanicalParams* defaultInstance();
        SOFA_CORE_API ExecParams* asExecParams(sofa::core::MechanicalParams*);
        SOFA_CORE_API const ExecParams* asExecParams(const sofa::core::MechanicalParams*);

        SOFA_CORE_API SReal kFactor(const sofa::core::MechanicalParams*);
        SOFA_CORE_API SReal bFactor(const sofa::core::MechanicalParams*);
        SOFA_CORE_API SReal kFactorIncludingRayleighDamping(const sofa::core::MechanicalParams*, SReal d);
        SOFA_CORE_API SReal mFactorIncludingRayleighDamping(const sofa::core::MechanicalParams*, SReal d);
        SOFA_CORE_API SReal dt(const sofa::core::MechanicalParams*);
    }
}

namespace sofa::core::visual::visualparams
{
    SOFA_CORE_API VisualParams* defaultInstance();

    SOFA_CORE_API ExecParams* asExecParams(sofa::core::visual::VisualParams*);
    SOFA_CORE_API const ExecParams* asExecParams(const sofa::core::visual::VisualParams*);

    SOFA_CORE_API sofa::core::visual::DrawTool* getDrawTool(VisualParams* params);
    SOFA_CORE_API sofa::core::visual::DisplayFlags& getDisplayFlags(VisualParams* params);
    SOFA_CORE_API sofa::core::visual::DrawTool* getDrawTool(const VisualParams* params);
    SOFA_CORE_API const sofa::core::visual::DisplayFlags& getDisplayFlags(const VisualParams* params);
}
```

The PR didn't changed existing simulation behavior or API and thus:
- shouldn't have any impact on sofa simulations
- shouldn't requires code refactoring

Because of the cleaning of the header dependency graph, third party code to include explicity the header files they are using.
The following ones are less visible:
```cpp
#include<sofa/core/ExecParams.h>
#include<sofa/core/ConstraintParams.h>
#include<sofa/core/MechanicalParams.h> 
#include<sofa/core/visual/VisualParams.h>
#include<sofa/core/visual/DisplayFlags.h>
#include<sofa/core/visual/DrawTool.h>
```

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

damienmarchal commented:
[ci-build][with-scene-tests]
Regression needs to be updated. 

PR to update Regression is: https://github.com/sofa-framework/regression/pull/6

damienmarchal committed:
[SofaCore] Add forward declarations for parameters.

damienmarchal committed:
[SofaSimulationCore] Add forward declaration.

damienmarchal committed:
[all] Update all the code base to use forward declaration & opaque api.

damienmarchal committed:
Merge remote-tracking branch 'upstream/master' into xp-remove-opaque-execparams

# Conflicts:
#	SofaKernel/modules/SofaBaseCollision/src/SofaBaseCollision/BruteForceDetection.cpp

damienmarchal committed:
[SofaBaseMechanics] Updates two tests to use execparams::defaultInstance()

damienmarchal commented:
A PR updating the Regression_test has been submitted https://github.com/sofa-framework/regression/pull/8

guparan commented on SofaKernel/modules/SofaSimulationCore/src/sofa/simulation/fwd.h:
```diff
@@ -55,3 +55,8 @@ SOFA_SIMULATION_CORE_API sofa::core::objectmodel::BaseContext* toBaseContext(Nod
 SOFA_SIMULATION_CORE_API Node* getNodeFrom(sofa::core::objectmodel::Base*);
 SOFA_SIMULATION_CORE_API Node* getNodeFrom(sofa::core::objectmodel::BaseContext*);
 }
+
+namespace sofa::simulation::common
+{
+    class MechanicalOperations;
+}
```
Should be in SofaSimulationCommon module?

damienmarchal replied:
Despite its namespace MechanicalOperations.h is in SofaSimulationCore. So until this is fixed it seems ok have its in this SoaSimulationCore/fwd.h

guparan commented on applications/plugins/PreassembledMass/PreassembledMass.inl:
```diff
@@ -57,7 +57,7 @@ void PreassembledMass< DataTypes >::bwdInit()
     {
         // perform assembly
-        core::MechanicalParams mparams = *core::MechanicalParams::defaultInstance();
+        core::MechanicalParams mparams = *core::GetDefaultMechanicalParamsInstance();
```
Should be `sofa::core::mechanicalparams::defaultInstance()` ?

damienmarchal replied:
No, 

EDIT: well maybe yes. I'm checking.
EDIT: Yes,

guparan commented on SofaKernel/modules/SofaCore/src/sofa/core/Mapping.h:
```diff
@@ -145,7 +145,7 @@ class Mapping : public BaseMapping
     /// This method generally corresponds to a symmetric stiffness matrix, but with rotations (which are not a commutative group) it is not the case.
     /// Since some solvers (including the Conjugate Gradient) require symmetric matrices, a flag is set in the MechanicalParams to say if symmetric matrices are required. If so, non-symmetric geometric stiffness should not be applied.
-    void applyDJT(const MechanicalParams* /*mparams = MechanicalParams::defaultInstance()*/ , MultiVecDerivId /*parentForce*/, ConstMultiVecDerivId  /*childForce*/ ) override;
+    void applyDJT(const MechanicalParams* /*mparams = GetDefaultMechanicalParamsInstance()*/ , MultiVecDerivId /*parentForce*/, ConstMultiVecDerivId  /*childForce*/ ) override;
```
Should be `sofa::core::mechanicalparams::defaultInstance()` ?

damienmarchal replied:
For the commented code, I prefer much more to remove them.

guparan commented on SofaKernel/modules/SofaCore/src/sofa/core/Mapping.inl:
```diff
@@ -224,7 +224,7 @@ void Mapping<In,Out>::applyJT(const ConstraintParams* cparams, MultiMatrixDerivI
 
 template <class In, class Out>
-void Mapping<In,Out>::applyDJT(const MechanicalParams* /*mparams = MechanicalParams::defaultInstance()*/ , MultiVecDerivId /*parentForce*/, ConstMultiVecDerivId  /*childForce*/ )
+void Mapping<In,Out>::applyDJT(const MechanicalParams* /*mparams = GetDefaultMechanicalParamsInstance()*/ , MultiVecDerivId /*parentForce*/, ConstMultiVecDerivId  /*childForce*/ )
```
Should be `sofa::core::mechanicalparams::defaultInstance()` ?

guparan commented on SofaKernel/modules/SofaSimulationCore/src/sofa/simulation/AnimateVisitor.cpp:
```diff
@@ -69,7 +69,7 @@ void AnimateVisitor::fwdInteractionForceField(simulation::Node*, core::behavior:
 
     MultiVecDerivId   ffId      = VecDerivId::externalForce();
-    MechanicalParams mparams; // = MechanicalParams::defaultInstance();
+    MechanicalParams mparams; // = GetDefaultMechanicalParamsInstance();
```
Should be `sofa::core::mechanicalparams::defaultInstance()` ?

guparan commented on SofaKernel/modules/SofaSimulationCore/src/sofa/simulation/MechanicalMatrixVisitor.h:
```diff
@@ -273,7 +272,7 @@ class SOFA_SIMULATION_CORE_API MechanicalApplyProjectiveConstraint_ToMatrixVisit
     const sofa::core::behavior::MultiMatrixAccessor* matrix;
 
-    MechanicalApplyProjectiveConstraint_ToMatrixVisitor(const core::MechanicalParams* mparams /* PARAMS FIRST  = core::MechanicalParams::defaultInstance()*/, const sofa::core::behavior::MultiMatrixAccessor* _matrix )
+    MechanicalApplyProjectiveConstraint_ToMatrixVisitor(const core::MechanicalParams* mparams /* PARAMS FIRST  = core::GetDefaultMechanicalParamsInstance()*/, const sofa::core::behavior::MultiMatrixAccessor* _matrix )
```
Should be `sofa::core::mechanicalparams::defaultInstance()` ?

guparan commented on SofaKernel/modules/SofaSimulationCore/src/sofa/simulation/Visitor.cpp:
```diff
@@ -33,7 +33,7 @@ Visitor::Visitor(const core::ExecParams* p)
     , params(p)
 {
-    //params = core::MechanicalParams::defaultInstance();
+    //params = core::GetDefaultMechanicalParamsInstance();
```
Should be `sofa::core::mechanicalparams::defaultInstance()` ?

guparan commented on applications/plugins/BulletCollisionDetection/BulletRigidContactMapper.h:
```diff
@@ -104,8 +104,8 @@ class BulletRigidContactMapper : public BaseContactMapper<DataTypes>
-            map->apply(core::MechanicalParams::defaultInstance(), core::VecCoordId::position(), core::ConstVecCoordId::position());
-            map->applyJ(core::MechanicalParams::defaultInstance(), core::VecDerivId::velocity(), core::ConstVecDerivId::velocity());
+            map->apply(core::GetDefaultMechanicalParamsInstance(), core::VecCoordId::position(), core::ConstVecCoordId::position());
+            map->applyJ(core::GetDefaultMechanicalParamsInstance(), core::VecDerivId::velocity(), core::ConstVecDerivId::velocity());
```
Should be `sofa::core::mechanicalparams::defaultInstance()` ?

guparan commented on applications/plugins/BulletCollisionDetection/BulletRigidContactMapper.h:
```diff
@@ -114,7 +114,7 @@ class BulletRigidContactMapper : public BaseContactMapper<DataTypes>
         {
             core::BaseMapping* map = mapping.get();
-            map->apply(core::MechanicalParams::defaultInstance(), core::VecCoordId::freePosition(), core::ConstVecCoordId::freePosition());
+            map->apply(core::GetDefaultMechanicalParamsInstance(), core::VecCoordId::freePosition(), core::ConstVecCoordId::freePosition());
```
Should be `sofa::core::mechanicalparams::defaultInstance()` ?

guparan commented on applications/plugins/Compliant/Compliant_gui/CompliantAttachPerformer.inl:
```diff
@@ -324,8 +324,8 @@ void CompliantAttachPerformer<DataTypes>::execute()
-//    mouseMapping->apply(core::MechanicalParams::defaultInstance());
-//    mouseMapping->applyJ(core::MechanicalParams::defaultInstance());
+//    mouseMapping->apply(core::GetDefaultMechanicalParamsInstance());
+//    mouseMapping->applyJ(core::GetDefaultMechanicalParamsInstance());
```
Should be `sofa::core::mechanicalparams::defaultInstance()` ?

guparan commented on applications/plugins/Compliant/contact/CompliantContact.h:
```diff
@@ -92,7 +92,7 @@ class CompliantContact : public BaseCompliantConstraintContact<TCollisionModel1,
     {
 
-//        simulation::MechanicalPropagateOnlyPositionAndVelocityVisitor bob( sofa::core::MechanicalParams::defaultInstance() );
+//        simulation::MechanicalPropagateOnlyPositionAndVelocityVisitor bob( sofa::core::GetDefaultMechanicalParamsInstance() );
```
Should be `sofa::core::mechanicalparams::defaultInstance()` ?

guparan commented on modules/SofaBoundaryCondition/SofaBoundaryCondition_test/PlaneForceField_test.cpp:
```diff
@@ -31,7 +31,7 @@ using sofa::simulation::setSimulation ;
 using sofa::core::objectmodel::BaseData ;
 using sofa::core::objectmodel::New ;
-using sofa::core::ExecParams ;
+using sofa::core::execparams::defaultInstance; /// From sofa/core/fwd.h
```
This Doxygen comment has no effect.

guparan commented on modules/SofaConstraint/SofaConstraint_test/BilateralInteractionConstraint_test.cpp:
```diff
@@ -26,12 +26,13 @@
 #include <SofaConstraint/BilateralInteractionConstraint.h>
 #include <SofaBaseMechanics/MechanicalObject.h>
 #include <sofa/core/MechanicalParams.h>
+using sofa::core::execparams::defaultInstance; /// From sofa/core/fwd.h
```
This Doxygen comment has no effect.

guparan commented on modules/SofaConstraint/src/SofaConstraint/GenericConstraintCorrection.cpp:
```diff
@@ -28,7 +28,7 @@
 #include <sofa/core/ObjectFactory.h>
 #include <sofa/core/behavior/ConstraintSolver.h>
-
+using sofa::core::execparams::defaultInstance; /// From sofa/core/fwd.h
```
This Doxygen comment has no effect.

guparan commented on modules/SofaExporter/SofaExporter_test/MeshExporter_test.cpp:
```diff
@@ -40,7 +40,7 @@ using sofa::simulation::Node ;
 #include <SofaSimulationCommon/SceneLoaderXML.h>
 using sofa::simulation::SceneLoaderXML ;
-using sofa::core::ExecParams ;
+using sofa::core::execparams::defaultInstance; /// From sofa/core/fwd.h
```
This Doxygen comment has no effect.

guparan commented on modules/SofaExporter/SofaExporter_test/OBJExporter_test.cpp:
```diff
@@ -40,7 +40,7 @@ using sofa::simulation::Node ;
 #include <SofaSimulationCommon/SceneLoaderXML.h>
 using sofa::simulation::SceneLoaderXML ;
-using sofa::core::ExecParams ;
+using sofa::core::execparams::defaultInstance; /// From sofa/core/fwd.h
```
This Doxygen comment has no effect.

guparan commented on SofaKernel/modules/SofaBaseCollision/SofaBaseCollision_test/DefaultPipeline_test.cpp:
```diff
@@ -25,6 +25,9 @@ using std::vector;
 using std::string;
 
+#include <sofa/core/fwd.h>
+using sofa::core::execparams::defaultInstance; /// From sofa/core/fwd.h
```
This Doxygen comment has no effect.

damienmarchal replied:
Actually I use /// to indicates code's comment that should stay in the code base from 
temporary comments with //

I'm not sure it makes sense to expect a doxygen comment here.

guparan commented on SofaKernel/modules/SofaBaseCollision/SofaBaseCollision_test/Sphere_test.cpp:
```diff
@@ -45,7 +46,7 @@ using sofa::component::collision::MeshMinProximityIntersection;
 using sofa::component::collision::MeshNewProximityIntersection ;
 
-using sofa::core::ExecParams ;
+using sofa::core::execparams::defaultInstance; /// From sofa/core/fwd.h
```
This Doxygen comment has no effect.

guparan commented on SofaKernel/modules/SofaBaseUtils/SofaBaseUtils_test/RequiredPlugin_test.cpp:
```diff
@@ -5,7 +5,7 @@
 using sofa::simulation::Node ;
 
-using sofa::core::ExecParams;
+using sofa::core::execparams::defaultInstance; /// From sofa/core/fwd.h
```
This Doxygen comment has no effect.

guparan commented on SofaKernel/modules/SofaEngine/SofaEngine_test/BoxROI_test.cpp:
```diff
@@ -30,6 +30,8 @@ using std::string;
 #include <gtest/gtest.h>
 using testing::Types;
 
+using sofa::core::execparams::defaultInstance; /// From sofa/core/fwd.h
```
This Doxygen comment has no effect.

guparan commented on SofaKernel/modules/SofaSimpleFem/SofaSimpleFem_test/TetrahedronFEMForceField_test.cpp:
```diff
@@ -32,7 +32,8 @@ using sofa::simulation::Node ;
 using sofa::simulation::SceneLoaderXML ;
-using sofa::core::ExecParams ;
+
+using sofa::core::execparams::defaultInstance; /// From sofa/core/fwd.h
```
This Doxygen comment has no effect.

guparan commented on applications/projects/runSofa/runSofaValidation.cpp:
```diff
@@ -23,6 +23,7 @@
 
 #include <sofa/helper/system/SetDirectory.h>
 using sofa::helper::system::SetDirectory;
+using sofa::core::execparams::defaultInstance; /// From sofa/core/fwd.h
```
This Doxygen comment has no effect.

guparan commented on modules/SofaBoundaryCondition/SofaBoundaryCondition_test/ConstantForceField_test.cpp:
```diff
@@ -50,8 +50,7 @@ using sofa::component::container::MechanicalObject ;
 using sofa::component::forcefield::ConstantForceField ;
-using sofa::core::ExecParams ;
-
+using sofa::core::execparams::defaultInstance; /// From sofa/core/fwd.h
```
This Doxygen comment has no effect.

guparan reviewed (changes requested):
I can't review all the files. :man_shrugging:

damienmarchal committed:
Merge remote-tracking branch 'upstream/master' into xp-remove-opaque-execparams

damienmarchal committed:
[all] Remove commented code

damienmarchal committed:
FIXUP

damienmarchal committed:
[BulletCollisionDetection] Use mechanicalparams::defaultInstance()

damienmarchal committed:
[plugins/PreassembledMass] Use mechanicalparams::defaultInstance()

damienmarchal commented:
Hi @guparan,

Thanks for the careful review. 
Here are the changes:
- I fixed the erroneous GetMechanical... in plugins
- I removed all the comment code in /* mparams = xxxx */ because these lines makes no sense neither before the PR or after. 
- The /// is just the normal code comment and I have not expectation for them to generate any doxygen comment.

fredroy commented:
[ci-build][with-all-tests]

guparan commented:
`///` means Doxygen comment. It is misleading to use it with "no expectation for them to generate any doxygen comment".

damienmarchal commented:
> `///` means Doxygen comment. It is misleading to use it with "no expectation" for Doxygen to parse it.
Isn't there plenty of cases that use /// and don't generate any doxygen documentation ? 
To me /// or ///< means this is long term documentation (no specifically doxygen) while // is temporary comment. 

Now about our coding style it says: 
```
Most used commands are /// or /** before blocks and ///< after member declarations.
Please DO NOT use // or /// after member declarations.
``` 

As #include is not a declaration what is wrong in using /// or ///< ? Te fact  doxygen don't know what to do with non-declaration related documentation does not means it is not useful for ones that read the code to have them ? 

So you have one more topic for tomorrow meeting :)

guparan commented:
When I see a Doxygen comment at a non-Doxygen-standard place, I always wonder "Why is this here? Does Doxygen now support this?". 
How can it be more useful to use `///` than `//` ?

damienmarchal commented:
> When I see a Doxygen comment at a non-Doxygen-standard place, I always wonder "Why is this here? Does Doxygen now support this?".
> How can it be more useful to use `///` than `//` ?

I understand your point and here is a summary to boostrap discussion. 

Currently in our code base we never esthablish specific rules on the use of doxygen comments and we are in fact mixing two different approaches. Let's name them "code centric" and "doxygen centric". 

In the code centric approach: 
-  /// or ///< is used to state that this the comment *is part of the documentation*. 
-  // is for any other *non-documenting* comment (eg: commented code). 
With this approach the main focus is that the documentation parts should be easily visible from the source code and that documentation is composed of declaration parts related part (that must fullfill doxygen rules) as well as internal details.   
The added value of using /// or ///< is that developper when they see them clearly knows "this is part of the documentation". This aproach is a super-set of doxygen but is not limiting the documentation to only what doxygen understand (the underlying assumption is that we need to document much more than just declaration/api). 

The doxygen centric approach is slighly different: 
- // should be used for internal-documentation as well as non-documentating comments
- /// or ///< should be used only for documentation what can be understood by Doxgen to generate html pages. 
Te main focus with this approach is to make very clear what will be use by Doxygen what it will use to auto-generate its public documentation, the counter part I see is that it mix comments and what should be considered as internal documentation. 

It is important is of course to have doxygen friendly declaration centric (the public API) documentation ...but it is equally important to have proper documentation for the internal details. 
If would probably do something like that in a new project:
- always use /**  */ for documenting public API (what should and must be visible in doxygen, mostly the comment close to declaration) 
- always use /// or ///< for internal details (no requirement that it must be close to declaration) 
- use // or /* */ for other usages. 

But given our code base ... I'm not sure which direction we should go.

guparan commented:
Or we can use the Doxygen style for comments that are **meant to** appear on [the SOFA Doxygen](https://www.sofa-framework.org/api) and the C++ style for the rest.
I don't think we need a complex policy on this.

damienmarchal commented:
So much time for something that does not shows in any case in the doxygen docs. I gave up, I will fix that. 

EDIT: So now this PR is clean on that aspect.

damienmarchal committed:
[all] Remove the Doxygen style comments "/// From sofa/core/fwd.h"

damienmarchal committed:
[all] Rename asExecParam to dynamicCastToExecParam (to make it constant with other opaque cast nameing)

damienmarchal commented:
[ci-build][with-scene-tests][with-regression-tests]

damienmarchal committed:
Merge remote-tracking branch 'upstream/master' into xp-remove-opaque-execparams

# Conflicts:
#	SofaKernel/modules/SofaCore/src/sofa/core/fwd.cpp
#	SofaKernel/modules/SofaCore/src/sofa/core/fwd.h
#	applications/plugins/SofaSimpleGUI/VisualPickVisitor.cpp
#	modules/SofaUserInteraction/src/SofaUserInteraction/InteractionPerformer.h

guparan commented on SofaKernel/modules/SofaCore/src/sofa/core/fwd.cpp:
```diff
@@ -23,8 +23,8 @@ const ConstraintParams* defaultInstance()
-ExecParams* asExecParams(sofa::core::ConstraintParams* cparams){ return cparams; }
-const ExecParams* asExecParams(const sofa::core::ConstraintParams* cparams){ return cparams; }
+ExecParams* dynamicCastToExecParams(sofa::core::ConstraintParams* cparams){ return cparams; }
+const ExecParams* dynamicCastToExecParams(const sofa::core::ConstraintParams* cparams){ return cparams; }
```
dynamicCast functions without using `dynamic_cast` ?

damienmarchal replied:
lol you are right. As it is a direct parent there is no need for a cast here.

damienmarchal replied:
So I finally swtich to the following convention:
castTo<AnyType>(Base*)
castToBase(aSubTypeOfBase)
castToExecParams(ASubTypeOfExecParam)...

damienmarchal committed:
Merge branch 'master' into xp-remove-opaque-execparams

damienmarchal committed:
[all] Unify all name for cast function.

The names are:
MyObject* castTo<MyObject>(Base*);
Base* castToBase(MyType*);
ExecParams* castToExecParams(OtherParam);

damienmarchal committed:
[TO REVERT] Switch SofaPython3 version for the one integrating #PR122 [TO REVERT]

damienmarchal commented:
Hi all,
 the name were changed, 
the pull request for sofapython3 is https://github.com/sofa-framework/SofaPython3/pull/122
the CI "pass" (at least as well as the master branch)

guparan commented on applications/plugins/SofaPython3/ExternalProjectConfig.cmake.in:
```diff
@@ -2,8 +2,8 @@ cmake_minimum_required(VERSION 3.11)
-    GIT_REPOSITORY https://github.com/sofa-framework/SofaPython3
-    GIT_TAG origin/master
+    GIT_REPOSITORY https://github.com/CRIStAL-PADR/plugin.SofaPython3.git
+    GIT_TAG origin/pr-update-for-sofa-pr-1794
```
```suggestion
    GIT_REPOSITORY https://github.com/sofa-framework/SofaPython3
    GIT_TAG ce09dd0fb5e11c1a333693234dc79d314d697657
```

damienmarchal committed:
Update applications/plugins/SofaPython3/ExternalProjectConfig.cmake.in

Co-authored-by: Guillaume Paran <guillaume.paran@sofa-framework.org>

alxbilger commented:
@damienmarchal  I just pulled these changes, and I have issues to compile.
I still found `sofa::core::ExecParams::defaultInstance()` instead of `sofa::core::execparams::defaultInstance()`. After changing them, I still have compilation errors:

[error.txt](https://github.com/sofa-framework/sofa/files/6140943/error.txt)

EDIT: I reset the SofaPython3 folder and the compilation worked
